aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1308 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+  <section name="section0">
+    <key name="key0">value</key>
+    <key name="key1">value</key>
+  </section>
+  <section name="section1">
+    <key name="key0">value</key>
+    <key name="key1">value</key>
+  </section>
+</configuration>
+```
 
-Typical ASP.NET Core 2.x apps use the static convenience method `CreateDefaultBuilder` to build the host:
+The previous configuration file loads the following keys with `value`:
 
-[!code-csharp[](index/sample_snapshot//Program.cs?highlight=12)]
+* section:section0:key:key0
+* section:section0:key:key1
+* section:section1:key:key0
+* section:section1:key:key1
 
-`CreateDefaultBuilder` loads optional configuration from *appsettings.json*, *appsettings.{Environment}.json*, [user secrets](xref:security/app-secrets) (in the `Development` environment), environment variables, and command-line arguments. The CommandLine configuration provider is called last. Calling the provider last allows the command-line arguments passed at runtime to override configuration set by the other configuration providers called earlier.
+Attributes can be used to supply values:
 
-For *appsettings* files where:
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+  <key attribute="value" />
+  <section>
+    <key attribute="value" />
+  </section>
+</configuration>
+```
 
-* `reloadOnChange` is enabled.
-* Contain the same setting in the command-line arguments and an *appsettings* file.
-* The *appsettings* file containing the matching command-line argument is changed after the app starts.
+The previous configuration file loads the following keys with `value`:
 
-If all the preceding conditions are true, the command-line arguments are overridden.
+* key:attribute
+* section:key:attribute
 
-ASP.NET Core 2.x app can use [WebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilder) instead of `CreateDefaultBuilder`. When using `WebHostBuilder`, manually set configuration with [ConfigurationBuilder](/api/microsoft.extensions.configuration.configurationbuilder). See the ASP.NET Core 1.x tab for more information.
+::: moniker range=">= aspnetcore-2.1"
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x/)
+## Key-per-file Configuration Provider
 
-Create a [ConfigurationBuilder](/api/microsoft.extensions.configuration.configurationbuilder) and call the `AddCommandLine` method to use the CommandLine configuration provider. Calling the provider last allows the command-line arguments passed at runtime to override configuration set by the other configuration providers called earlier. Apply the configuration to [WebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilder) with the `UseConfiguration` method:
+The [KeyPerFileConfigurationProvider](/dotnet/api/microsoft.extensions.configuration.keyperfile.keyperfileconfigurationprovider) uses a directory's files as configuration key-value pairs. The key is the file name. The value contains the file's contents.
```

Comment:
Why would you use this? The only reference I can find refers to docker.

Reply:
> Why would you use this?

**_FEVER!_** 🚑 lol

I actually didn't know ... and didn't ask. It's a real configuration provider tho. Are you sure it should be stripped out? It's not part of the sample, so it would be easy to :hocho: it out.

Reply:
It doesn't have to be removed, but its use is not obvious and deserves some mention of the expected scenario.

Reply:
Ok ... will add quick comment that it's used in Docker hosting scenarios on the next commit.